ui: Match Databases Tab to current designs

mrtracy opened the pull request:
Commit overhauls the database tab to be more inline with the current designs.
+ Removed "Database" page to display details on a single database. Instead,
database details (such as tables and grants) are displayed directly on the
databases list page.
+ Updated table details page to match current designs.
+ Created shared "SummaryBar" component, which can be used to implement a
common motif in new UI designs - a group of summarized statistics.

Still to do:

+ Move pages into new outer navigation
+ Change sub-page navigation (currently between "tables" and "grants") to a
select dropdown instead of tabs
+ Better match styling of current designs (i.e. adjusting fonts, colors)
+ Better column widths for "grant" tables
+ Schema change information
+ Syntax highlighting for create table statements

Screenshots
![screen shot 2016-11-08 at 3 56 42 pm](https://cloud.githubusercontent.com/assets/6969858/20116884/1d64fb28-a5cc-11e6-95ed-35bfb0388fe1.png)

![screen shot 2016-11-08 at 3 56 50 pm](https://cloud.githubusercontent.com/assets/6969858/20116886/2222c2da-a5cc-11e6-9452-6fc08f214eb6.png)

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/10552)
<!-- Reviewable:end -->

Commit:
ui: Match Databases Tab to current designs

Commit overhauls the database tab to be more inline with the current designs.
+ Removed "Database" page to display details on a single database. Instead,
database details (such as tables and grants) are displayed directly on the
databases list page.
+ Updated table details page to match current designs.
+ Created shared "SummaryBar" component, which can be used to implement a
common motif in new UI designs - a group of summarized statistics.

Still to do:

+ Move pages into new outer navigation
+ Change sub-page navigation (currently between "tables" and "grants") to a
select dropdown instead of tabs
+ Better match styling of current designs (i.e. adjusting fonts, colors)
+ Better column widths for "grant" tables
+ Schema change information
+ Syntax highlighting for create table statements

maxlang commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

If you could break up databaseSummary.tsx it would be a bit easier to parse.

Otherwise looks good!

---

Review status: 0 of 13 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[pkg/ui/app/components/summaryBar.tsx, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9dl9O0rbgv_OzifnD:-KW9dl9PK1wph7swa8py:b-rizw23) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/app/components/summaryBar.tsx#L37)):*
> ```XML
>       <div className="summary-value">{this.computeValue()}</div>
>       <div className="summary-title">{this.props.title}</div>
>     </div>;
> ```

Random thought: How much do you care about semantic HTML? This feels like a good match for a definition list.

https://css-tricks.com/utilizing-the-underused-but-semantically-awesome-definition-list/

---

*[pkg/ui/app/containers/databases/databases.tsx, line 19 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9eW4LIIIhJeVMnFtr:-KW9eW4MLU2vC5SuAct8:b6he439) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/app/containers/databases/databases.tsx#L19)):*
> ```XML
> // with our backend methods.
> // TODO(mrtracy): This exclusion should occur on the backend methods, which
> // should handle virtual tables correctly.
> ```

Issue #9689

---

*[pkg/ui/app/containers/databases/databaseSummary.tsx, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9iyLutiv6br6mMQZp:-KW9iyLv9cbFqGr27IZq:b-5c84xz) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/app/containers/databases/databaseSummary.tsx#L1)):*
> ```XML
> import _ from "lodash";
> ```

Maybe split up this mega file somehow?

---

*[pkg/ui/styl/partials/layout.styl, line 705 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9heM6cGqKyHDlrq1r:-KW9heM6cGqKyHDlrq1s:bpo9ogq) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/styl/partials/layout.styl#L705)):*
> ```Stylus
> 
> .section.databases
>   max-width 1350px
> ```

Does the page do wonky things when you shrink the browser width?

---

*[pkg/ui/styl/partials/layout.styl, line 726 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9gofuWFdHqiiH1HUw:-KW9gofuWFdHqiiH1HUx:b-z1fa1y) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/styl/partials/layout.styl#L726)):*
> ```Stylus
> 
>     .summary-value
>       font-size 30px
> ```

Huh, this CSS is all surprisingly succinct. Nice.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-:-KW9lk_ERTojrydfWuvP:b20qp6q)*
<!-- Sent from Reviewable.io -->

mrtracy commented:
Review status: 0 of 14 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[pkg/ui/app/components/summaryBar.tsx, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9dl9O0rbgv_OzifnD:-KW9yYolxDY_qYHwuckQ:bnxm0iv) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/app/components/summaryBar.tsx#L37)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> Random thought: How much do you care about semantic HTML? This feels like a good match for a definition list.
> 
> https://css-tricks.com/utilizing-the-underused-but-semantically-awesome-definition-list/
</details>
We can check that out later, for now lets just keep them all in react components and we can change the underlying html later.

---

*[pkg/ui/app/containers/databases/databases.tsx, line 19 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9eW4LIIIhJeVMnFtr:-KW9yejVAbTNnsvlmnvu:b-7jgylk) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/app/containers/databases/databases.tsx#L19)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> Issue #9689
</details>
Added a note.

---

*[pkg/ui/app/containers/databases/databaseSummary.tsx, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9iyLutiv6br6mMQZp:-KW9yRMCB0ICKu94Zkm_:bq2im47) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/app/containers/databases/databaseSummary.tsx#L1)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> Maybe split up this mega file somehow?
</details>
Split up into three files - databaseSummary, databaseTables and databaseGrants.  Each one is more focused.

---

*[pkg/ui/styl/partials/layout.styl, line 705 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552#-KW9heM6cGqKyHDlrq1r:-KW9qd-ATi2jkNKAcqBp:b27h6uz) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0df89ddcd4cf74fb6a1099dc63edafd72bee247/pkg/ui/styl/partials/layout.styl#L705)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> Does the page do wonky things when you shrink the browser width?
</details>
this is a max width, it prevents it from looking weird by taking up the whole page on a widescreen display.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10552)*
<!-- Sent from Reviewable.io -->
